asf_public_discourse_home_decarbonisation/pipeline/faqs_identification/faq_identification.py

```diff
@@ -0,0 +1,208 @@
+    else:
+        topic_model = topic_model_definition(min_topic_size, representation_model)
+    topics, probs = topic_model.fit_transform(docs)
+    topics_, topics_info, doc_info = get_outputs_from_topic_model(topic_model, docs)
+
+    logger.info(f"Number of topics: {len(topics_info) - 1}")
+    logger.info(
+        f"% of outliers: {topics_info[topics_info['Topic'] == -1]['%'].values[0]}"
+    )
+
+    if reduce_outliers_to_zero:
+        new_topics = topic_model.reduce_outliers(docs, topics, strategy="embeddings")
+        topic_model.update_topics(docs, topics=new_topics)
+        topics, topics_info, doc_info = get_outputs_from_topic_model(topic_model, docs)
+        topics_info.sort_values("Count", ascending=False)
+
+        logger.info(f"Number of topics: {len(topics_info) - 1}")
+        logger.info(
+            f"% of outliers: {topics_info[topics_info['Topic'] == -1]['%'].values[0]}"
+        )
+
+    topics_info = update_topics_with_duplicates(topics_info, doc_info, questions_data)
+    doc_info = update_docs_with_duplicates(doc_info, questions_data)
+
+    return topics_info, doc_info
+
+
+if __name__ == "__main__":
+    # Dealing with user defined arguments
+    args = parse_arguments()
+    source = args.source
+    reduce_outliers_to_zero = args.reduce_outliers_to_zero
+    filter_by_expression = args.filter_by_expression
+    start_date = args.start_date
+    end_date = args.end_date
+    min_topic_size = args.min_topic_size
+    openai_key = args.openai_key
+
+    sentences_data = pd.read_csv(
+        f"s3://{S3_BUCKET}/data/{source}/outputs/topic_analysis/{source}_{filter_by_expression}_{start_date}_{end_date}_sentences_data.csv",
+    )
+
+    # Extract sentences which are question
+    sentences_data["is_question"] = sentences_data["sentences"].str.contains("\?")
+    questions_data = sentences_data[sentences_data["is_question"]]
+
+    docs = list(questions_data.drop_duplicates("sentences")["sentences"])
+    dates = list(questions_data.drop_duplicates("sentences")["date"])
+
+    # Identifying topics without using OpenAI's representation model
+    topics_info, doc_info = run_faq_identification(
+        min_topic_size=min_topic_size,
+        docs=docs,
+        questions_data=questions_data,
+        reduce_outliers_to_zero=reduce_outliers_to_zero,
+    )
+
+    topics_info.to_csv(
+        f"s3://{S3_BUCKET}/data/{source}/outputs/topic_analysis/{source}_{filter_by_expression}_{start_date}_{end_date}_FAQ_topics_info_no_representation.csv",
+        index=False,
+    )
+    doc_info.to_csv(
+        f"s3://{S3_BUCKET}/data/{source}/outputs/topic_analysis/{source}_{filter_by_expression}_{start_date}_{end_date}_FAQ_docs_info_no_representation.csv",
+        index=False,
+    )
+
+    # Identifying topics using LLMs representation model
+    # the topics should be the same as above, but a different representation model is used
+    representation_model = OpenAI(
+        openai_key,
+        model="gpt-3.5-turbo",
```

Comment:
```suggestion
        model="gpt-4o-mini",
```
Updated to the latest 'cheap' model